### DISCUSSION TITLE
Is `useFetcher` memorized?

### DISCUSSION THREAD

**Discussion by mattstobbs** [remix-run/remix#1276] [Q&A] [answered]
Hi,

In the [code](https://github.com/kentcdodds/kentcdodds.com/blob/388084ff3c2bb9c4a71fcea90b35264b90fb70ac/app/utils/theme-provider.tsx#L48-L53) for Kent C. Dodds' blog, when he uses `useFetcher`, he has a `TODO` comment:

```js
const persistTheme = useFetcher()
// TODO: remove this when persistTheme is memoized properly
const persistThemeRef = React.useRef(persistTheme)
React.useEffect(() => {
  persistThemeRef.current = persistTheme
}, [persistTheme])
```

I was wondering whether this is still needed when using `useFetcher`? Or is it being memorized properly now?

**Comment by meglio** [comment]
Any updates on this?

**Comment by sergiodxa** [comment, marked as answer]
As it can be seen on the code the result object of useFetcher is memoized.

https://github.com/remix-run/remix/blob/cfb614be3d457fd9742bd98aaac2f196008e6c58/packages/remix-react/components.tsx#L1424-L1454

**Comment by meglio** [comment]
@mattstobbs , so this issue can now be considered resolved?

**Comment by mattstobbs** [comment]
Looks like it, thanks everyone!
